webapi/src/main/scala/org/knora/webapi/slice/admin/domain/service/KnoraGroupService.scala

```diff
@@ -21,6 +33,33 @@ case class KnoraGroupService(knoraGroupRepo: KnoraGroupRepo) {
   def findById(id: GroupIri): Task[Option[KnoraGroup]] = knoraGroupRepo.findById(id)
 
   def findByIds(ids: Seq[GroupIri]): Task[Chunk[KnoraGroup]] = knoraGroupRepo.findByIds(ids)
+
+  def createGroup(request: GroupCreateRequest, project: Project): Task[KnoraGroup] =
+    for {
+      _        <- ensureGroupNameIsUnique(request.name)
+      _        <- ensureProjectExists(ProjectIri.unsafeFrom(project.id))
+      groupIri <- iriService.checkOrCreateNewGroupIri(request.id, Shortcode.unsafeFrom(project.shortcode))
+      group =
+        KnoraGroup(
+          id = groupIri,
+          groupName = request.name,
+          groupDescriptions = request.descriptions,
+          status = request.status,
+          belongsToProject = Some(ProjectIri.unsafeFrom(project.id)),
+          hasSelfJoinEnabled = request.selfjoin,
+        )
+      _ <- knoraGroupRepo.save(group)
+    } yield group
```

Comment:
Shouldn't this belong in KnoraGroupRepo? What's the difference between KnoraGroupRepo and this file? In master this file is a really thin wrapper over it, does it really serve a purpose? I see @seakayone made it.

Reply:
The sole responsibility for repositories  is to store entities and retrieve them. Maintaining constraints on certain properties of the entity which cannot be modelled by a type system, e.g. the uniqueness of a property other than the id of the entity is not the responsibility of a repository.

The `admin` module does not expose the repositories but only services operating on these. The methods of these services represent certain use cases and contain the business logic based on them, e.g. the creation of a group with a custom IRI.